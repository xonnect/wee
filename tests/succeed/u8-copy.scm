;; test for the 2 cases of vector copy
;; TODO error case ?
(define x '#u8(0 1 2 3 4 5))
(define y (u8vector 10 11 12 13 14 15 16 17 18 19))
(define z (u8vector 20 21 22 23 24 25 26 27 28 29))

(define (display-vector v)
  (let loop ((i 0))
    (if (< i (u8vector-length v))
	(begin (display (u8vector-ref v i))
	       (display " ")
	       (loop (+ i 1)))))
  (display "\n"))

;; (display-vector x)
;; (display-vector y)
;; (display-vector z)

(u8vector-copy! x 1 y 3 4)
(u8vector-copy! y 2 z 4 5)
(u8vector-copy! z 0 z 2 2)
(display-vector x)
(display-vector y)
(display-vector z)
