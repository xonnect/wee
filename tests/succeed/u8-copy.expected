0 1 2 3 4 5 
10 11 12 1 2 3 4 17 18 19 
20 21 20 21 12 1 2 3 4 29 
